src/services/TaskService.ts

```diff
@@ -0,0 +1,78 @@
+import { IHttpResponse } from '@rocket.chat/apps-engine/definition/accessors';
+import { IUser } from '@rocket.chat/apps-engine/definition/users';
+
+import { TodoistApp } from '../../TodoistApp';
+import { getTaskUrl, getTasksUrl } from '../helpers/const';
+import { ITask } from '../interfaces/tasks';
+
+export class TaskService {
+  constructor(private app: TodoistApp) {}
+
+  public async fetch(user: IUser, taskId?: string): Promise<ITask[]> {
+    try {
+      const url = taskId ? getTaskUrl(taskId) : getTasksUrl();
+      const response: IHttpResponse = await this.app.getHttpHelperInstance().get(user, url);
+      if (response.statusCode !== 200) {
+        throw new Error(`Failed to fetch tasks: ${response.content}`);
+      }
+      const tasks: ITask[] = Array.isArray(response.data) ? response.data : [response.data];
```

Comment:
I think you should be a little less optimistic here.
you cannot know what the type is. So what I would do is something like this:


```suggestion
      this.validate(response.data);

      ...

    private validate(tasks: unknown): void {
        if (tasks === null || tasks === undefined || !Array.isArray(tasks)) {
            throw new Error(`Invalid task structure: ${JSON.stringify(task)}`);
        }

        tasks.forEach((task) => {
            if (typeof task !== object || task === null) {
                throw new Error(`Invalid task structure: ${JSON.stringify(task)}`);
            }

            if (!('id' in task) || typeof task.id !== string) {
                throw new Error(`Invalid task structure: ${JSON.stringify(task)}`);
            }

            if (!('content' in task) || typeof task.content !== string) {
                throw new Error(`Invalid task structure: ${JSON.stringify(task)}`);
            }
        });
    }

```